tools/dune_rule_gen/path.ml

```diff
@@ -0,0 +1,50 @@
+(************************************************************************)
+(* This file is licensed under The MIT License                          *)
+(* See LICENSE for more information                                     *)
+(************************************************************************)
+
+(* We distinguish Path arguments as to adjust for subdirs in Coq
+   theory evaluation *)
+
+type t = Rel of string | Abs of string
+
+let to_string = function Rel p -> p | Abs p -> p
+
+let make path = if Filename.is_relative path then Rel path else Abs path
+
+let map ~f = function Rel p -> Rel (f p) | Abs p -> Abs (f p)
+
+let fold ~f = function Rel p -> f p | Abs p -> f p
+
+let relative dir path = map dir ~f:(fun dir -> Filename.concat dir path)
+
+let gen_sub n = String.concat Filename.dir_sep @@ List.init n (fun _ -> "..")
+
+let adjust ~lvl = function
+  | Rel path -> Rel (Filename.concat (gen_sub lvl) path)
+  | Abs path -> Abs path
+
+let compare p1 p2 = match p1, p2 with
+  | Rel p1, Rel p2 -> String.compare p1 p2
+  | Abs p1, Abs p2 -> String.compare p1 p2
+  | Rel _, Abs _ -> -1
+  | Abs _, Rel _ -> 1
+
+let pp fmt = function
+  | Rel p -> Format.fprintf fmt "%s" p
+  | Abs p -> Format.fprintf fmt "%s" p
+
+let basename = fold ~f:Filename.basename
+
+let add_extension ~ext = map ~f:(fun p -> p ^ ext)
+
+let replace_ext ~ext = map ~f:(fun p -> Filename.remove_extension p ^ ext)
+
+(* Coqdep strips files with dirname "." of the basename, so we need
```

Comment:
@ejgallego could you open an issue about this

Reply:
This is already discussed in https://github.com/coq/coq/pull/16116 , and that's due to makefile woes. I'm not interesting in fixing it for now.

Reply:
https://github.com/coq/coq/pull/16116